src/utils.ts

```diff
@@ -33,6 +33,25 @@ export function addUrlQuery(url: string, query: Record<string, unknown>): string
   return parsed.toString();
 }
 
+/**
+ * Checks if the provided bigint can fit in a 64-bit unsigned integer.
+ *
+ * @param int - The provided integer.
+ * @throws - Will throw if the int does not fit in 64 bits.
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN | MDN Demo}
+ */
+export function checkUint64(int: bigint) {
+  const maxint = BigInt("18446744073709551615"); // max uint64
```

Comment:
this could be defined on top of the file, otherwise we build the big integer each time this function is called - minor but low hanging fruit perf optimisation